pkg/cloud/vsphere/services/cloudprovider/csi.go

```diff
@@ -562,6 +564,7 @@ func CSICloudConfigSecret(data string) *corev1.Secret {
 func ConfigForCSI(ctx *context.ClusterContext) *cloudprovider.Config {
 	config := &cloudprovider.Config{}
 
+	config.Global.ClusterID = fmt.Sprintf("%s/%s", ctx.Cluster.Namespace, ctx.Cluster.Name)
```

Comment:
Hi @andrewsykim,

You probably want to be careful here. If `ctx.Cluster.Namespace` is not set then you likely want to use `v1.DefaultNamespace` from K8s. Or just use the string `"default"`.

Reply:
Hi @andrewsykim,

Also, either check for this to see if it's empty (and don't set it unless it is), or let's figure out a way to hide it from being assigned externally. As it is you've got as part of the CRD, and it shouldn't be if you're setting it at runtime no matter the value a user may have assigned.

Reply:
I don't think the namespace on a namespaced object be empty though

Reply:
Okay, I guess the API server sets it to `default`. Either way, the config field shouldn't be public then.

Reply:
Which config field?

Reply:
`ClusterID`

Reply:
That needs to be exported so we can set it from the cloudprovider package though.

Reply:
Spoke offline with Andrew, because `ClusterID` gets propagated as a field in `VSphereCluster`, at the very least we need a comment on the field that says it shouldn't be set by any other consumer of the API